pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -1050,10 +1028,10 @@ impl<T: AsRef<DeltaLayerInner>> ValueRef<T> {
     }
 }
 
-struct Adapter<T: AsRef<DeltaLayerInner>>(T);
+pub(crate) struct Adapter<T: AsRef<DeltaLayerInner>>(T);
 
-impl<T: AsRef<DeltaLayerInner>> BlockReader for Adapter<T> {
-    fn read_blk(&self, blknum: u32) -> Result<BlockLease, std::io::Error> {
+impl<T: AsRef<DeltaLayerInner>> Adapter<T> {
+    pub(crate) fn read_blk(&self, blknum: u32) -> Result<BlockLease, std::io::Error> {
         self.0.as_ref().file.read_blk(blknum)
     }
 }
```

Comment:
Because this is ok, then it must be ok to keep the old versions, but it need to add the `'a` to ValueRef like it is right now, but also add the `T` from `Adapter<T>` which needs to be on `BlockCursor<'a, T>`.

Reply:
Not needed anymore.